core/src/impl/Kokkos_Error.cpp

```diff
@@ -25,8 +25,21 @@
 #include <Kokkos_Core.hpp>  // show_warnings
 #include <impl/Kokkos_Error.hpp>
 
+#ifndef KOKKOS_COMPILER_MSVC
+#define KOKKOS_IMPL_BUILTIN_UNREACHABLE() __builtin_unreachable()
+#else
+#define KOKKOS_IMPL_BUILTIN_UNREACHABLE() \
+  do {                                    \
+  } while (false)
```

Comment:
Now that I look at this again I wonder if that is not potentially confusing as one might think I had written an infinite loop as a hint to the compiler that it cannot be reached but my actual intent was to have some dummy code that needs to be followed by an underscore.
Maybe I should do `static_assert(true, "no-op to require trailing semicolon")` instead.  Thoughts?

Reply:
We are using `__builtin_unreachable` in a bunch of other places so it makes sense to me to define it in a more central place.

Reply:
https://en.cppreference.com/w/cpp/utility/unreachable shows that you could try using
```
__assume(false)
```
for MSVC instead.

Reply:
Does not belong to this work